docs/source/API/core/MultiGPUSupport.rst

```diff
@@ -0,0 +1,153 @@
+    int N = ...;
+
+    // Create stream on device N
+    cudaStream_t stream;
+    cudaSetDevice(N);
+    cudaStreamCreate(&stream);
+
+    // Scope execution space to ensure stream
+    // is destroyed *after* execution space
+    {
+      // Create execution space
+      Kokkos::Cuda exec_space(stream);
+
+      // Use execution space
+      /* ... */
+    }
+
+    // Destroying stream after use
+    cudaSetDevice(N);
+    cudaStreamDestroy(stream);
+
+HIP
+~~~
+
+For the HIP backend, like with CUDA, the user creates a ``hipStream_t`` object and passes it to the ``Kokkos::HIP`` constructor.
+
+.. note:: The user is expected to manage the lifetime of the stream. Only after the execution space is destroyed
+          can the user safely destroy the stream.
+
+.. warning:: Multi-GPU only supported for ROCm 5.6 and later. Because of the lack of HIP API functions for querying a
+             stream's device before ROCm 5.6, constructing a ``Kokkos::HIP`` instance on a non-default device isn't
+             supported, and problems will likely arise downstream.
+
+
+.. code-block:: cpp
+
+    // Query number of devices available
+    int n_devices;
+    hipGetDeviceCount(&n_devices);
+
+    // Choose some 0 <= N < n_devices
+    int N = ...;
+
+    // Create stream on device N
+    hipStream_t stream;
+    hipSetDevice(N);
+    hipStreamCreate(&stream);
+
+    // Scope execution space to ensure stream
+    // is destroyed *after* execution space
+    {
+      // Create execution space
+      Kokkos::HIP exec_space(stream);
+
+      // Use execution space
+      /* ... */
+    }
+
+    // Destroying stream after use
+    hipSetDevice(N);
+    hipStreamDestroy(stream);
+
+SYCL
+~~~~
+
+For the SYCL backend, the user creates a ``sycl::queue`` object and passes it to the ``Kokkos::SYCL`` constructor.
+
+.. code-block:: cpp
+
+    // Get list of devices available
+    std::vector<sycl::device> gpu_devices =
+      sycl::device::get_devices(sycl::info::device_type::gpu);
+
+    // Choose some 0 <= N < gpu_devices.size()
+    int N = ...;
+
+    // Create a queue on device N.
+    // Note: Kokkos requires SYCL queues to be "in_order"
+    sycl::queue queue{gpu_devices[N], sycl::property::queue::in_order()};
+
+    // Create execution space
+    Kokkos::SYCL exec_space(queue);
+
+    // Use execution space
+    /* ... */
+
+Using Kokkos Methods
+--------------------
+
+Once an execution space has been created on the chosen device, the execution space must be passed to all Kokkos methods
+intended to be used on device. In the absence of an execution space, Kokkos will use the execution space on the default
+device.
+
+Allocating Managed Views
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+To allocate a managed view on device, pass the execution space to ``Kokkos::view_alloc()``.
+
+Example:
+
+.. code-block:: cpp
+
+    using ExecutionSpace = decltype(exec_space);
+    Kokkos::View<int*, ExecutionSpace> V(Kokkos::view_alloc("V", exec_space), 10);
```

Comment:
```suggestion
    auto V(Kokkos::view_alloc("V", exec_space), 10);
```

Reply:
yeah, you are right. But this should work:
https://godbolt.org/z/MhY5n3Pbq

Reply:
I'd rather be explicit about the template arguments (and I like passing memory spaces as template arguments to `Kokkos::View` instead of execution spaces but I don't think that's crucial here).